platform/src/components/aws/redis.ts

```diff
@@ -334,23 +422,31 @@ export class Redis extends Component implements Link.Linkable {
     const version = all([engine, args.version]).apply(
       ([engine, v]) => v ?? (engine === "redis" ? "7.1" : "7.2"),
     );
-    const instance = output(args.instance).apply((v) => v ?? "t4g.micro");
-    const argsCluster = normalizeCluster();
-    const vpc = normalizeVpc();
 
     const dev = registerDev();
     if (dev?.enabled) {
       this.dev = dev;
       return;
     }
 
+    const vpc = normalizeVpc();
+    const serverless = normalizeServerless();
+    if (serverless.enabled) {
```

Comment:
since `normalizeServerless` returns an `Output`, the `if (serverless.enabled)` expression always evalutates to true, thus always creating a serverless cluster